src/main/webapp/app/exam/participate/summary/exam-participation-summary.component.ts

```diff
@@ -50,10 +56,17 @@ export class ExamParticipationSummaryComponent implements OnInit {
         // courseId is not part of the exam or the exercise
         this.courseId = Number(this.route.snapshot.paramMap.get('courseId'));
         this.setExamWithOnlyIdAndStudentReviewPeriod();
+        this.setIcon(this.exercise.type);
+    }
+
+    setIcon(exerciseType?: ExerciseType) {
+        if (exerciseType) {
+            this.icon = getIcon(exerciseType) as IconProp;
+        }
     }
 
-    getIcon(exerciseType: ExerciseType) {
-        return getIcon(exerciseType);
+    asProgrammingExercise(exercise: Exercise): ProgrammingExercise {
+        return exercise as ProgrammingExercise;
```

Comment:
if anyone give me idea how to tackle this issue without casting like this, that would be great :)

Reply:
@iozbeyli told me that if you use it once, and the component doesn't constantly update, then it should be okay to use. I'm not 100% sure though :D

Reply:
Don't blame for it :smile:

Reply:
If it is not inside a for loop etc put it into a separate variable and add a setter for exercise that puts the casted version into a ProgrammingExercise variable so that you can use it